File: app/scripts/seed.ts
/**
 * Seeds one agency, one client, and a usable strategy.
 *
 * Usage:
 *   npm run seed -- --email you@example.com [--uid <firebase-uid>]
 *
 * Pass --uid to bind an existing Firebase account as the agency admin. Without
 * it the agency is created but has no members, and the first person to sign in
 * claims it (see ensureMember in lib/auth.ts).
 *
 * It writes a populated content quota, which now paces SCHEDULING rather than
 * generation — an empty one is workable. What the planner actually needs is an
 * active campaign with a content plan: with none it refuses to run, because a
 * campaign's plan is the demand.
 */
import { config } from "dotenv";
import { resolve } from "path";

config({ path: resolve(process.cwd(), ".env.local") });

async function main() {
  const args = process.argv.slice(2);
  const emailIdx = args.indexOf("--email");
  const uidIdx = args.indexOf("--uid");
  const email = emailIdx >= 0 ? args[emailIdx + 1] : null;
  const uid = uidIdx >= 0 ? args[uidIdx + 1] : null;

  // Imported after dotenv so the Admin SDK sees the credentials.
  const { db, COLLECTIONS, FieldValue } = await import("../lib/firestore");

  // Attach to the agency that already exists rather than refusing. The common
  // case is seeding AFTER signing in, when sign-in has already created the
  // agency and made you its admin — refusing there would mean the seed could
  // never add test data.
  const existing = await db().collection(COLLECTIONS.agencies).limit(1).get();
  let agencyId: string;

  if (!existing.empty) {
    agencyId = existing.docs[0].id;
    console.log(`Agency:  ${agencyId} (existing)`);
  } else {
    const agencyRef = db().collection(COLLECTIONS.agencies).doc();
    await agencyRef.set({
      name: email ? `${email.split("@")[0]}'s agency` : "My agency",
      createdAt: FieldValue.serverTimestamp(),
    });
    agencyId = agencyRef.id;
    console.log(`Agency:  ${agencyId} (created)`);
  }

  if (uid) {
    await db().collection(COLLECTIONS.members).doc(uid).set({
      agencyId,
      email: email ?? null,
      name: null,
      role: "admin",
      createdAt: FieldValue.serverTimestamp(),
    });
    console.log(`Member:  ${uid} (admin)`);
  } else {
    const members = await db().collection(COLLECTIONS.members).limit(1).get();
    console.log(
      members.empty
        ? "Member:  none yet — the first user to sign in claims this agency as admin."
        : "Member:  left as is."
    );
  }

  const clientRef = db().collection(COLLECTIONS.clients).doc();
  await clientRef.set({
    agencyId,
    name: "Example Client",
    websiteUrl: null,
    logoUrl: null,
    description: "Seeded client. Edit or delete.",
    contactEmail: null,
    contactPhone: null,
    timezone: "America/New_York",
    branding: null,
    status: "active",
    googleCalendarId: null,
    createdAt: FieldValue.serverTimestamp(),
    updatedAt: FieldValue.serverTimestamp(),
  });
  console.log(`Client:  ${clientRef.id}`);

  await db().collection(COLLECTIONS.strategies).doc(clientRef.id).set({
    clientId: clientRef.id,
    businessName: "Example Client",
    icp: {
      description: "Define your ideal customer profile",
      // Every key DEMOGRAPHIC_FIELDS renders, so the ICP tab opens with a form
      // rather than a blank grid the operator has to guess the shape of.
      demographics: {
        customer_type: "",
        industry: "",
        role: "",
        location: "",
        age_range: "",
        language: "",
        company_size: "",
        income_or_revenue: "",
      },
      pain_points: [],
      desired_outcomes: [],
      objections: [],
      trigger_events: [],
    },
    voice: {
      personality: "Define your brand personality",
      traits: [],
      tone: "conversational, direct, practical",
      communication_style: "",
      words_to_use: [],
      words_to_avoid: [
        "revolutionary",
        "game-changing",
        "crushing it",
        "leverage",
        "synergy",
        "unlock your potential",
      ],
    },
    // primary_angle / secondary_angles, NOT the `angles` this used to seed —
    // nothing has ever read `angles`, so the seeded positioning was invisible
    // in the editor.
    positioning: {
      primary_angle: { type: "", statement: "", why: "" },
      secondary_angles: [],
      anti_positioning: "",
      differentiation: "",
    },
    messaging: { tagline: "", value_props: [], key_messages: [], proof_points: [] },
    goals: { primary: "awareness", secondary: "leads", focus_90_days: "", metrics: [] },
    contentStrategy: { platforms: [], content_pillars: [] },
    // Publishable formats only. This used to seed `hook` and `cta`, which are
    // retired (see content-types.ts) — the quota table flags them and tells you
    // to delete the row, so the seed was writing rows to be undone.
    contentQuota: {
      weekly: {
        post: { count: 3, channels: ["linkedin"] },
        carousel: { count: 1, channels: ["linkedin", "instagram"] },
        newsletter: { count: 1, channels: ["email"] },
      },
      rationale: "Seeded default. Adjust in Strategy > Content Quota.",
    },
    createdAt: FieldValue.serverTimestamp(),
    updatedAt: FieldValue.serverTimestamp(),
  });
  console.log("Strategy: seeded with a populated weekly quota.");
}

main()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });

File: app/db/seed.py
"""Seed script to insert initial agency, user, client, strategy, and daily task config."""

import asyncio
import uuid
from datetime import datetime

from sqlalchemy import select

from app.db.models import Agency, Client, Strategy, TaskConfig, User
from app.db.session import async_session

# Use fixed UUIDs matching migration defaults
DEFAULT_AGENCY_ID = uuid.UUID("00000000-0000-0000-0000-000000000001")
DEFAULT_USER_ID = uuid.UUID("00000000-0000-0000-0000-000000000002")
DEFAULT_CLIENT_ID = uuid.UUID("00000000-0000-0000-0000-000000000003")


async def seed():
    async with async_session() as db:
        # Ensure default agency exists
        result = await db.execute(select(Agency).where(Agency.id == DEFAULT_AGENCY_ID))
        if not result.scalar_one_or_none():
            db.add(Agency(id=DEFAULT_AGENCY_ID, name="Default Agency"))
            print("Agency seeded.")

        # Ensure default user exists
        result = await db.execute(select(User).where(User.id == DEFAULT_USER_ID))
        if not result.scalar_one_or_none():
            db.add(User(
                id=DEFAULT_USER_ID,
                agency_id=DEFAULT_AGENCY_ID,
                email="admin@agency.local",
                name="Admin",
                role="admin",
            ))
            print("User seeded.")

        # Ensure default client exists
        result = await db.execute(select(Client).where(Client.id == DEFAULT_CLIENT_ID))
        if not result.scalar_one_or_none():
            db.add(Client(
                id=DEFAULT_CLIENT_ID,
                agency_id=DEFAULT_AGENCY_ID,
                name="Default Client",
                status="active",
            ))
            print("Client seeded.")

        await db.flush()

        # Check if strategy already exists for default client
        result = await db.execute(
            select(Strategy).where(Strategy.client_id == DEFAULT_CLIENT_ID)
        )
        if result.scalar_one_or_none():
            print("Strategy already exists, skipping seed.")
        else:
            strategy = Strategy(
                id=uuid.uuid4(),
                client_id=DEFAULT_CLIENT_ID,
                business_name="Your Business Name",
                icp={
                    "description": "Define your ideal customer profile",
                    "demographics": {},
                    "pain_points": [],
                    "goals": [],
                },
                voice={
                    "personality": "Define your brand personality",
                    "traits": [],
                    "words_to_use": [],
                    "words_to_avoid": [
                        "revolutionary",
                        "game-changing",
                        "crushing it",
                        "leverage",
                        "synergy",
                        "unlock your potential",
                    ],
                    "tone": "conversational, direct, practical",
                },
                positioning={
                    "angles": [
                        {
                            "type": "contrarian",
                            "description": "Challenge what everyone believes",
                        },
                        {
                            "type": "unique_mechanism",
                            "description": "Lead with HOW, not WHAT",
                        },
                        {
                            "type": "transformation",
                            "description": "Before and after, the gap closed",
                        },
                    ],
                    "anti_positioning": "What you are NOT",
                },
                messaging={
                    "value_props": [],
                    "key_messages": [],
                    "tagline": "",
                },
                goals={
                    "primary": "awareness",
                    "secondary": "leads",
                    "metrics": [],
                },
                created_at=datetime.utcnow(),
                updated_at=datetime.utcnow(),
            )
            db.add(strategy)
            print("Strategy seeded.")

        # Check if daily task config exists
        result = await db.execute(
            select(TaskConfig).where(TaskConfig.task_type == "daily")
        )
        if result.scalar_one_or_none():
            print("Daily task config already exists, skipping seed.")
        else:
            daily_config = TaskConfig(
                id=uuid.uuid4(),
                agency_id=DEFAULT_AGENCY_ID,
                task_type="daily",
                name="Daily Marketing Run",
                description="Generates daily marketing content: 1 post, 2 hooks, 1 CTA, 1 campaign idea, 1 recommended action.",
                pipeline=[
                    {"step": "context_build", "config": {"lookback_days": 14}},
                    {"step": "planning", "config": {"prompt": "planning_daily"}},
                    {"step": "generation", "config": {"prompt": "generation_daily"}},
                    {"step": "review", "config": {"prompt": "review_daily", "max_retries": 2}},
                    {"step": "post_production", "config": {"prompt": "post_production_daily", "image_variants": 2}},
                    {"step": "schedule_suggest", "config": {}},
                    {"step": "log_run", "config": {}},
                    {"step": "deliver_slack", "config": {}},
                ],
                schedule="0 8 * * *",
                active=True,
            )
            db.add(daily_config)
            print("Daily task config seeded.")

        await db.commit()
        print("Seed complete.")


if __name__ == "__main__":
    asyncio.run(seed())
